jdaviz/configs/imviz/plugins/conesearch/conesearch.py

```diff
@@ -0,0 +1,55 @@
+from astropy.table import Table
+from astropy.coordinates import SkyCoord
+from astroquery.sdss import SDSS
+
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin, ViewerSelectMixin
+from jdaviz.configs.imviz.helper import get_top_layer_index
+
+__all__ = ['ConeSearch']
+
+
+@tray_registry('imviz-conesearch', label="Imviz ConeSearch")
+class ConeSearch(PluginTemplateMixin, ViewerSelectMixin):
+    template_file = __file__, "conesearch.vue"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def vue_do_conesearch(self, *args, **kwargs):
+        # current viewer object
+        curr_viewer = self.viewer.selected_obj
+
+        # used to obtain the current image visible in the viewer
+        viewer = self.app.get_viewer_by_id(self.viewer_selected)
+        i = get_top_layer_index(viewer)
+        data = viewer.state.layers[i].layer
+
+        # obtains the center point of the current image and converts the point into sky coordinates
+        center_point = curr_viewer._get_zoom_center(data)
+        x_center = center_point[0]
+        y_center = center_point[1]
+        skycoord_center = curr_viewer.state.reference_data.coords.pixel_to_world(x_center, y_center)
+
+        # obtains the viewer's zoom limits (just one) based on the visible layer
+        zoom_limits = self.viewer.selected_obj._get_zoom_limits(data)
+        zoom_x_limit = zoom_limits[0, 0]
+        zoom_y_limit = zoom_limits[0, 1]
+        zoom_coordinate = curr_viewer.state.reference_data.coords.pixel_to_world(zoom_x_limit, zoom_y_limit)
```

Comment:
You want data from all four corners, sometimes the WCS can be asymmetrical, especially with distortion involved. But you also don't want coordinates beyond the image corners if viewer is zoomed out too much.

```suggestion
        ny, nx = curr_viewer.state.reference_data.shape
        zoom_x_min = max(0, curr_viewer.state.x_min)
        zoom_x_max = min(nx, curr_viewer.state.x_max)
        zoom_y_min = max(0, curr_viewer.state.y_min)
        zoom_y_max = min(ny, curr_viewer.state.y_max)
        zoom_coordinate = curr_viewer.state.reference_data.coords.pixel_to_world(
            [zoom_x_min, zoom_x_min, zoom_x_max, zoom_x_max],
            [zoom_y_min, zoom_y_max, zoom_y_max, zoom_y_min])
```